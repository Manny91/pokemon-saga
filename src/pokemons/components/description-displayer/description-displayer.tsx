import { LoadingWrapper, Loading } from "../loading/loading";
import styled from "../../../styled-components";
import React from "react";

export const PokemonDescriptionDisplayer = ({
  description,
  loading,
}: PokemonDetails) => {
  return (
    <DescriptionDisplay>
      {!loading && <Description>{description}</Description>}
      {loading && <Loading />}
    </DescriptionDisplay>
  );
};

const DescriptionDisplay = styled.div`
  border: 1px solid;
  background-color: ${(props) => props.theme.colors.greenScreen};
  padding: 7px;
  border-radius: 3px;

  border: inset ${(props) => props.theme.colors.greenBorder} 3px;
  ${LoadingWrapper} {
    margin: 7px;
  }
`;
interface PokemonDetails {
  description: string;
  loading: boolean;
}
const Description = styled.h4`
  margin: 0px;
  overflow: scroll;
  height: 100%;
`;
