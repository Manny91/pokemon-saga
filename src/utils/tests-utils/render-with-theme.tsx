import React from "react";
import { ThemeProvider } from "../../styled-components";
import { theme } from "../../theme";
import { render } from "@testing-library/react";

export const renderWithTheme = (ui: JSX.Element) => {
  return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
};
