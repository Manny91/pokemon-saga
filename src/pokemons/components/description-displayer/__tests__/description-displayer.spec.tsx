import React from "react";
import { PokemonDescriptionDisplayer } from "../description-displayer";
import "jest-styled-components";
import { renderWithTheme } from "../../../../utils/tests-utils/render-with-theme";

describe("PokemonDescriptionDisplayer Component", () => {
  it("should render properly when loading is false", () => {
    const testProps = {
      loading: false,
      description: "pokemon description test"
    };
    const subject = renderWithTheme(
      <PokemonDescriptionDisplayer {...testProps} />
    );
    const tree = subject.container;
    expect(tree).toMatchSnapshot();
  });

  it("should render properly when loading is true", () => {
    const testProps = {
      loading: true,
      description: "pokemon description test"
    };
    const subject = renderWithTheme(
      <PokemonDescriptionDisplayer {...testProps} />
    );
    const tree = subject.container;
    expect(tree).toMatchSnapshot();
  });
});
