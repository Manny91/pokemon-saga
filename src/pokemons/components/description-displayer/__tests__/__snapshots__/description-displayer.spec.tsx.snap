// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDescriptionDisplayer Component should render properly when loading is false 1`] = `
.c0 {
  border: 1px solid;
  background-color: #A5CD53;
  padding: 7px;
  border-radius: 3px;
  border: inset #9aa28b 3px;
}

.c0 .sc-AxirZ {
  margin: 7px;
}

.c1 {
  margin: 0px;
  overflow: scroll;
  height: 100%;
}

<div>
  <div
    class="c0"
  >
    <h4
      class="c1"
    >
      pokemon description test
    </h4>
  </div>
</div>
`;

exports[`PokemonDescriptionDisplayer Component should render properly when loading is true 1`] = `
.c4 {
  width: 4px;
  height: 4px;
  background-color: black;
  border-radius: 1px;
  margin: 0px 2px;
  list-style-type: none;
  -webkit-animation: fZpxSO 1s infinite ease-in-out both;
  -webkit-animation: fZpxSO 1s infinite ease-in-out both;
  animation: fZpxSO 1s infinite ease-in-out both;
}

.c2 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0px;
}

.c2 .c3:nth-child(1) {
  -webkit-animation-delay: -0.35s;
  animation-delay: -0.35s;
}

.c2 .c3:nth-child(2) {
  -webkit-animation-delay: -0.2s;
  animation-delay: -0.2s;
}

.c0 {
  border: 1px solid;
  background-color: #A5CD53;
  padding: 7px;
  border-radius: 3px;
  border: inset #9aa28b 3px;
}

.c0 .c1 {
  margin: 7px;
}

<div>
  <div
    class="c0"
  >
    <ul
      class="c1 c2"
    >
      <li
        class="c3 c4"
      />
      <li
        class="c3 c4"
      />
      <li
        class="c3 c4"
      />
    </ul>
  </div>
</div>
`;
